### ISSUE TITLE
[trace on] TypeError: 'NoneType' object is not callable

### ISSUE THREAD

**Issue by anki-code** [xonsh/xonsh#3446]
```
$ xonsh --version
xonsh/0.9.13

$ cat trace_except.xsh
trace on
echo 123

$ xonsh trace_except.xsh
trace_except.xsh:2:echo 123
123
Exception ignored in: <function _removeHandlerRef at 0x7fc4cea67d40>
Traceback (most recent call last):
  File "/home/pc/miniconda3/lib/python3.7/logging/__init__.py", line 766, in _removeHandlerRef
  File "/home/pc/miniconda3/lib/python3.7/site-packages/xonsh/__amalgam__.py", line 19689, in trace
TypeError: 'NoneType' object is not callable
```

`miniconda3/lib/python3.7/site-packages/xonsh/__amalgam__.py", line 19689`:
```
    def trace(self, frame, event, arg):
        """Implements a line tracing function."""
        if event not in self.valid_events:
            return self.trace
        fname = find_file(frame)      ############ 19689
        if fname in self.files:
```

**Comment by scopatz** [comment]
hmmm weird.  Thanks for reporting! PRs welcome, of course!

**Comment by anki-code** [comment]
Solved in 0.9.22
